framework/DataObjects/XHistorySet.py

```diff
@@ -236,16 +232,20 @@ def _toCSV(self,fileName,start=0,**kwargs):
     # specialized to write custom RAVEN-style history CSVs
     # TODO some overlap with DataSet implementation, but not much.
     keep = self._getRequestedElements(kwargs)
+    toDrop = list(var for var in self._orderedVars if var not in keep)
     # don't rewrite everything; if we've written some already, just append (using mode)
-    if start > 0:
-      # slice data starting at "start"
-      sl = slice(start,None,None)
-      data = self._data.isel(**{self.sampleTag:sl})
-      mode = 'a'
+    mode = 'a' if start > 0 else 'w'
+    # hierarchical (get just the ending)
```

Comment:
The comment and logic here are a little confusing.  The comment says we're covering the hierarchical case, but we start the logic with "if not hierarchical".  Maybe extending the comment to explain that discrepancy would help the next dev through this section.

Reply:
added a comment